content/cloud-docs/agents/hooks.mdx

```diff
@@ -0,0 +1,137 @@
+---
+page_title: Hooks - Terraform Cloud Agents - Terraform Cloud and Terraform Enterprise
+---
+
+# Hooks
+
+Terraform Cloud Agents support executing custom programs, or hooks, during
+strategic points of a Terraform run. These hooks allow operators to extend the
+functionality of Terraform runs.
+
+Terraform Cloud Agents support the following hooks:
+
+- `pre-plan` - This hook executes before `terraform init` during a `plan`
+operation.
+
+- `post-plan` - This hook executes after `terraform plan` during a `plan`
+operation regardless whether or not `terraform plan` successfully executed.
+
+- `pre-apply` - This hook executes before `terraform init` during an `apply`
+operation.
+
+- `post-apply` - This hook executes after `terraform apply` during an `apply`
+operation regardless whether or not `terraform apply` successfully executed.
+
+## Limitations
+
+Please note the following when using hooks.
+
+- If the hook exits with a non-zero exit code, the Terraform run will fail
+immediately.
+- The name of the hook is not customizable.
+- The hook must be executable. That is, the hook must have execute permission.
+- The hook has a 60 second timeout.
+- Environment variables do not persist across hooks. For example, if a
+`pre-plan` hook exports environment variables, they will not be available
+during `terraform plan` nor will they be available during the `post-plan` hook.
+Similarly, if `terraform plan` exports environment variables, they will not be
+available during the `post-plan` hook.
+
+## Configuration
+
+Hooks are stored within the `hooks` subdirectory of the agent's data directory.
+By default, that's `~/.tfc-agent/hooks`.
+
+To enable a hook, create a file named `terraform-${HOOK}` where `${HOOK}` is
+one of the hooks listed above. Then store that file within the `hooks`
+subdirectory and make it executable. Once an agent is started and receives a
+run, it will attempt to execute your hook at the specified point during the
+Terraform run.
+
+For example, a `pre-plan` hook uses an executable file named
+`~/.tfc-agent/hooks/terraform-pre-plan`.
+
+### Configuring hooks when running an Agent as a binary
+
+When running the `tfc-agent` binary directly on a Linux instance, create the
+`hooks` directory within the agent's data directory.
+
+```sh
+mkdir -p ~/.tfc-agent/hooks
+```
+
+Then create your hooks. Here, a `pre-plan` hook is being created.
+
+```sh
+cat <<EOF > ~/.tfc-agent/hooks/terraform-pre-plan
+#!/bin/bash
+echo "Example pre-plan hook."
+EOF
+```
+
+Don't forget to make the hooks executable!
+
+```sh
+chmod +x ~/.tfc-agent/hooks/terraform-pre-plan
+```
+
+Finally, start the agent.
+
+```sh
+tfc-agent
+```
+
+That's it! As the agent executes Terraform runs it will attempt to execute
```

Comment:
I like this friendly tone! But, we should stay away from using exclamation points in our docs and things that make it seem like setup is easy/that we think it's easy. If a user has some type of error, we don't want to make them feel bad with our enthusiasm. :-) Also, I try to use different language for things like "execute" (where it makes sense!) because our style guide advises us to steer away from violent language. Sometimes it can be unavoidable (execute is a common term in programming and things are literally called "executables"), but in a lot of places here I think "run" could be a good substitute - what do you think about that?

Also you said "attempt to create your enabled hooks." Three things:
- Are there any like common failure modes or gotchas that would make this not work? Attempt made me nervous haha. If not we should sya "Terraform will run your enabled hooks." 
- If possible, can we say "Terraform will run your enabled hooks" - i don't want to make users feel unsure that it will work properly!
- when you say "enabled" - is there something else folks have to do to enable them? OR is it just that like if there's an executable run task in the folder, it's getting added? If the l latter is the case, I wonder if it makes more sense to say "existing".

Reply:
Updated!